src/python/vdo-setup-dmtest/vdo-setup-dmtest.py

```diff
@@ -0,0 +1,512 @@
+        if (key.upper().find('DMTEST') != -1):
+          sys.exit(1)
+        else:
+          self.log.warning("Manual install of {name} required".format(name = repoName))
+      finally:
+        os.chdir(RUN_DIR)
+
+      self.log.info("Completed installation of {0}\n".format(repoName))
+
+  def installBufioTest(self, repo):
+    """
+    Perform installation steps for the test-bufio repo
+    """
+    # Skip if module is already installed
+    moduleName = 'dm_bufio_test'
+    result = runCommandIgnoringErrors(['lsmod | grep', moduleName])
+    if (result.returncode == 0):
+      self.log.debug("Module {0} is already installed - skipping".format(moduleName))
+      return
+
+    # Obtain the list of remote OS-specific branches
+    # Bufio-test historically only has rhel-8 and rhel-9 branches
+    remoteBranches = []
+    for ref in repo.remotes.origin.refs:
+      if ("rhel" in ref.name):
+        remoteBranches.append(ref.name.replace('origin/', "", 1))
+    self.log.debug("Bufio remote OS-specific branches are: {0}".format(remoteBranches))
+
+    # Determine the rhel release branch to use for installation
+    if (os.path.exists('/etc/fedora-release')):
+      osVersion = 9
+    elif (os.path.exists('/etc/redhat-release')):
+      result = runCommand('cut -d " " -f6 /etc/redhat-release | cut -d"." -f1')
+      osVersion = result.stdout.rstrip()
+    elif (os.path.exists('/etc/centos-release')):
+      result = runCommand('cut -d " " -f4 /etc/centos-release')
+      osVersion = result.stdout.rstrip()
+
+    if (int(osVersion) > 9):
+      osVersion = 9
+
+    branch = 'rhel-' + str(osVersion)
+    assert branch in remoteBranches, " ".join(['Invalid remote branch', branch])
+
+    # Build and install the module for the running kernel
+    repo.git.switch(branch)
+
+    # Check for a module signing certificate/key pair, or create one
+    certsDir = os.path.join('/lib/modules', os.uname().release, 'build/certs')
+    certPath = os.path.join(certsDir, 'signing_key.pem')
+
+    self.log.info('Checking for signing certificate {0}'.format(certPath))
+    certExists = os.path.exists(certPath)
+    if (certExists):
+      certValid = verifyCertificate(certPath)
+
+    if not certExists or not certValid:
+      # Check for a x509.genkey file in the certs directory, or create one
+      genkeyPath = os.path.join(certsDir, 'x509.genkey')
+      if not (os.path.exists(genkeyPath)):
+        assert createX509Genkey(certsDir) == True, 'Unable to create x509.genkey'
+
+      self.log.info("Creating the certificate/key pair needed for signing the module")
+      keyPath = os.path.join(certsDir, 'signing_key.x509')
+      opensslCmd = ['sudo openssl req -new -nodes -utf8 -sha512 -days 36500 -batch -x509',
+                    '-config', genkeyPath, '-outform PEM -out', certPath, '-keyout', keyPath]
+      result = runCommand(opensslCmd)
+      assert result.returncode == 0, "Creating OpenSSL certificate failed"
```

Comment:
Since the other cert operations are in their own functions, perhaps this is also a candidate for moving into its own function?

Reply:
Fixed in latest push.